nav2_route/src/plugins/graph_file_loaders/geojson_graph_file_loader.cpp

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+
+#include <filesystem>
+#include <memory>
+#include <string>
+#include <vector>
+#include <fstream>
+
+#include "nav2_route/plugins/graph_file_loaders/geojson_graph_file_loader.hpp"
+
+namespace nav2_route
+{
+
+void GeoJsonGraphFileLoader::configure(
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr node)
+{
+  RCLCPP_INFO(node->get_logger(), "Configuring geojson graph file loader");
+  logger_ = node->get_logger();
+}
+
+bool GeoJsonGraphFileLoader::loadGraphFromFile(
+  Graph & graph, GraphToIDMap & graph_to_id_map, std::string filepath)
+{
+  if (!fileExists(filepath)) {
+    RCLCPP_ERROR(logger_, "Failed to find file at %s", filepath.c_str());
+    return false;
+  }
+
+  std::ifstream graph_file(filepath);
+
+  Json graph_geojson;
+  graph_file >> graph_geojson;
+
+  auto features = graph_geojson["features"];
+  std::vector<Json> nodes;
+  std::vector<Json> edges;
+  getGraphElements(features, nodes, edges);
+
+  if (nodes.empty()) {
+    RCLCPP_ERROR(logger_, "No nodes were found in %s", filepath.c_str());
+    return false;
+  }
+
+  if (edges.empty()) {
+    RCLCPP_ERROR(logger_, "No edges were found in %s", filepath.c_str());
+    return false;
+  }
+
+  graph.resize(nodes.size());
+  addNodesToGraph(graph, graph_to_id_map, nodes);
+  addEdgesToGraph(graph, graph_to_id_map, edges);
+
+  return true;
+}
+
+bool GeoJsonGraphFileLoader::fileExists(const std::string & filepath)
+{
+  return std::filesystem::exists(filepath);
+}
+
+
+void GeoJsonGraphFileLoader::getGraphElements(
+  const Json & features, std::vector<Json> & nodes, std::vector<Json> & edges)
+{
+  for (const auto & feature : features) {
+    if (feature["geometry"]["type"] == "Point") {
+      nodes.emplace_back(feature);
+    }
+    if (feature["geometry"]["type"] == "MultiLineString") {
+      edges.emplace_back(feature);
+    }
+  }
+}
+
+void GeoJsonGraphFileLoader::addNodesToGraph(
+  Graph & graph, GraphToIDMap & graph_to_id_map, std::vector<Json> & nodes)
+{
+  int idx = 0;
+  for (const auto & node : nodes) {
+    // Required data
+    unsigned int id = node["properties"]["id"];
+    std::string frame = node["properties"]["frame"];
+    float x = node["geometry"]["coordinates"][0];
+    float y = node["geometry"]["coordinates"][1];
+    graph[id].nodeid = id;
+    graph[id].coords.x = x;
+    graph[id].coords.y = y;
+    graph_to_id_map[graph[idx].nodeid] = idx;
+    idx++;
+  }
+}
+
+void GeoJsonGraphFileLoader::addEdgesToGraph(
+  Graph & graph, GraphToIDMap & graph_to_id_map, std::vector<Json> & edges)
+{
+  nav2_route::EdgeCost edge_cost;
+  for (const auto & edge : edges) {
+    // Required data
+    const auto edge_properties = edge["properties"];
+    unsigned int id = edge_properties["id"];
+    unsigned int start_id = edge_properties["startid"];
+    unsigned int end_id = edge_properties["endid"];
+
+    // Recommended data
+    if (edge_properties.contains("cost")) {
+      edge_cost.cost = edge_properties["cost"];
```

Comment:
EdgeCost isn't reset between iterations of the for loop, so if 1 populates it and others don't, that information is propogated. YOu might just want to allocate the `edge_cost` at the start of the for loop instead since its a trivial struct of a float and a bool (proportional to the things already being allocated for the unsigned int's).